src/NuGet.Services.AzureSearch/BaseDocumentBuilder.cs

```diff
@@ -175,16 +179,21 @@ public BaseDocumentBuilder(IOptionsSnapshot<AzureSearchJobConfiguration> options
 
             if (leaf.IconFile != null)
             {
-                var provider = new FlatContainerPackagePathProvider(_options.Value.FlatContainerContainerName);
-                var iconPath = provider.GetIconPath(document.PackageId, normalizedVersion);
-                document.IconUrl = new Uri(_options.Value.ParseFlatContainerBaseUrl(), iconPath).AbsoluteUri;
+                SetIconUrlFromFlatContainer(document);
             }
             else
             {
                 document.IconUrl = leaf.IconUrl;
             }
         }
 
+        private void SetIconUrlFromFlatContainer(IBaseMetadataDocument document)
```

Comment:
Maybe name it as "Get" method to keep the same convention as other items?

Reply:
The primary purpose of this method is to set document's icon URL (see `document.IconUrl = ...`), although it does have to fetch the icon URL as part of the method.

Reply:
Maybe define the "Get" function and use "document.IconUrl = GetIconUrlFromFlatContainer(***)" to align with other set-up? Just nit changes.

Reply:
I was thinking of doing that, but given that we're passing the `document` object to the function anyway, I felt that there wasn't too much point in passing it back to do the assignment.

Reply:
This file has the `Get*` pattern to set a property already, I agree with @zhhyu that style consistency would be preferable.